Parallel build doesn't work

BackupPC-XS-0.55

```
# uname -mrs
FreeBSD 11.0-RELEASE-p2 amd64
```

```
moiseev@esxi-moiseev:/home/moiseev/ports/src/BackupPC-XS-0.55# perl Makefile.PL
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for BackupPC::XS::md5
Writing MYMETA.yml and MYMETA.json
Generating a Unix-style Makefile
Writing Makefile for BackupPC::XS::zlib
Writing MYMETA.yml and MYMETA.json
Generating a Unix-style Makefile
Writing Makefile for BackupPC::XS
Writing MYMETA.yml and MYMETA.json
moiseev@esxi-moiseev:/home/moiseev/ports/src/BackupPC-XS-0.55# make -j2
--- blib/lib/BackupPC/.exists ---
--- blib/arch/.exists ---
--- blib/lib/auto/BackupPC/XS/.exists ---
--- blib/arch/auto/BackupPC/XS/.exists ---
--- blib/bin/.exists ---
--- blib/script/.exists ---
--- blib/man1/.exists ---
--- blib/man3/.exists ---
--- subdirs ---
--- subdirs ---
--- adler32.o ---
cc -c    -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_FORTIFY_SOURCE=2 -O2 -pipe -fstack-protector -fno-strict-aliasing    -DVERSION=\"\"  -DXS_VERSION=\"\" -DPIC -fPIC "-I/usr/local/lib/perl5/5.24/mach/CORE"   adler32.c
--- subdirs ---
--- md5.o ---
cc -c    -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_FORTIFY_SOURCE=2 -O2 -pipe -fstack-protector -fno-strict-aliasing    -DVERSION=\"\"  -DXS_VERSION=\"\" -DPIC -fPIC "-I/usr/local/lib/perl5/5.24/mach/CORE"   md5.c
In file included from md5.c:20:
./../rsync.h:222:22: error: typedef redefinition with different types ('unsigned int' vs '__id_t' (aka 'long'))
typedef unsigned int id_t;
                     ^
/usr/include/sys/types.h:138:17: note: previous definition is here
typedef __id_t          id_t;           /* can hold a uid_t or pid_t */
                        ^
In file included from md5.c:20:
./../rsync.h:228:22: error: typedef redefinition with different types ('unsigned int' vs '__mode_t' (aka 'unsigned short'))
typedef unsigned int mode_t;
                     ^
/usr/include/sys/types.h:158:18: note: previous definition is here
typedef __mode_t        mode_t;         /* permissions */
                        ^
In file included from md5.c:20:
./../rsync.h:236:22: error: typedef redefinition with different types ('unsigned int' vs '__size_t' (aka 'unsigned long'))
typedef unsigned int size_t;
                     ^
/usr/include/stdio.h:46:18: note: previous definition is here
typedef __size_t        size_t;
                        ^
In file included from md5.c:20:
./../rsync.h:290:3: error: Could not find a 32-bit integer variable
# error Could not find a 32-bit integer variable
  ^
In file included from md5.c:20:
In file included from ./../rsync.h:376:
./../md5/mdigest.h:10:8: error: expected ';' at end of declaration list
        uint32 A, B, C, D;
              ^
              ;
./../md5/mdigest.h:11:2: error: duplicate member 'int32'
        uint32 totalN;          /* bit count, lower 32 bits */
        ^
./../rsync.h:300:26: note: expanded from macro 'uint32'
# define uint32 unsigned int32
                         ^
./../md5/mdigest.h:10:2: note: previous declaration is here
        uint32 A, B, C, D;
        ^
./../rsync.h:300:26: note: expanded from macro 'uint32'
# define uint32 unsigned int32
                         ^
In file included from md5.c:20:
In file included from ./../rsync.h:376:
./../md5/mdigest.h:11:8: error: expected ';' at end of declaration list
        uint32 totalN;          /* bit count, lower 32 bits */
              ^
              ;
./../md5/mdigest.h:12:2: error: duplicate member 'int32'
        uint32 totalN2;         /* bit count, upper 32 bits */
        ^
./../rsync.h:300:26: note: expanded from macro 'uint32'
# define uint32 unsigned int32
                         ^
./../md5/mdigest.h:10:2: note: previous declaration is here
        uint32 A, B, C, D;
        ^
./../rsync.h:300:26: note: expanded from macro 'uint32'
# define uint32 unsigned int32
                         ^
In file included from md5.c:20:
In file included from ./../rsync.h:376:
./../md5/mdigest.h:12:8: error: expected ';' at end of declaration list
        uint32 totalN2;         /* bit count, upper 32 bits */
              ^
              ;
./../md5/mdigest.h:17:60: error: expected ')'
void mdfour_update(md_context *md, const uchar *in, uint32 length);
                                                           ^
./../md5/mdigest.h:17:19: note: to match this '('
void mdfour_update(md_context *md, const uchar *in, uint32 length);
                  ^
./../md5/mdigest.h:23:61: error: expected ')'
void md5_update(md_context *ctx, const uchar *input, uint32 length);
                                                            ^
./../md5/mdigest.h:23:16: note: to match this '('
void md5_update(md_context *ctx, const uchar *input, uint32 length);
               ^
md5.c:24:7: error: no member named 'A' in 'struct (anonymous at ./../md5/mdigest.h:9:9)'
        ctx->A = 0x67452301;
        ~~~  ^
md5.c:25:7: error: no member named 'B' in 'struct (anonymous at ./../md5/mdigest.h:9:9)'
        ctx->B = 0xEFCDAB89;
        ~~~  ^
md5.c:26:7: error: no member named 'C' in 'struct (anonymous at ./../md5/mdigest.h:9:9)'
        ctx->C = 0x98BADCFE;
        ~~~  ^
md5.c:27:7: error: no member named 'D' in 'struct (anonymous at ./../md5/mdigest.h:9:9)'
        ctx->D = 0x10325476;
        ~~~  ^
md5.c:29:7: error: no member named 'totalN' in 'struct (anonymous at ./../md5/mdigest.h:9:9)'
        ctx->totalN = ctx->totalN2 = 0;
        ~~~  ^
md5.c:29:21: error: no member named 'totalN2' in 'struct (anonymous at ./../md5/mdigest.h:9:9)'
        ctx->totalN = ctx->totalN2 = 0;
                      ~~~  ^
md5.c:34:8: error: expected ';' at end of declaration
        uint32 X[16], A, B, C, D;
              ^
              ;
md5.c:36:2: error: use of undeclared identifier 'A'
        A = ctx->A;
        ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
*** [md5.o] Error code 1

make[1]: stopped in /usr/home/moiseev/ports/src/BackupPC-XS-0.55/md5
1 error

make[1]: stopped in /usr/home/moiseev/ports/src/BackupPC-XS-0.55/md5
*** [subdirs] Error code 2

make: stopped in /usr/home/moiseev/ports/src/BackupPC-XS-0.55
--- subdirs ---
A failure has been detected in another branch of the parallel make

make[1]: stopped in /usr/home/moiseev/ports/src/BackupPC-XS-0.55/zlib
*** [subdirs] Error code 2

make: stopped in /usr/home/moiseev/ports/src/BackupPC-XS-0.55
2 errors

make: stopped in /usr/home/moiseev/ports/src/BackupPC-XS-0.55
moiseev@esxi-moiseev:/home/moiseev/ports/src/BackupPC-XS-0.55#
```

--- a/Makefile.PL
+++ b/Makefile.PL
@@ -29,6 +29,15 @@ WriteMakefile(
     clean             => { FILES => "config.log config.status config.h" },
 );
 
+# by Nicholas Clark <nick@ccl4.org>
+# https://github.com/Perl/perl5/commit/4d106cc5d8fd328d39b1db3c57572dd3dec915b5
+#
+# We don't want the default subdir rule, as it creates a race condition with the
+# rule we add below.
+sub MY::subdir_x {
+    return '';
+}
+
 sub MY::postamble {
         return <<'EOF';
 

